util/network-devp2p/src/discovery.rs

```diff
@@ -540,7 +555,7 @@ impl<'a> Discovery<'a> {
 		self.send_packet(PACKET_PONG, from, &response.drain())?;
 
 		let entry = NodeEntry { id: *node_id, endpoint: pong_to.clone() };
-		if !entry.endpoint.is_valid() {
+		if !entry.endpoint.is_valid_sync_node() && !bootnode {
```

Comment:
I think we could just use `is_valid_bootnode` here, right?

Reply:
`endpoint.is_valid_sync_node == endpoint.is_valid`

the extra check just asserts that if there was an error in getiing `NodeEndpoint` out of the packet, its a bootnode otherwise a peer sent us a bad address.

Reply:
Well if you think about it, what you are checking here is whether the received entry's endpoint is a _syncing node_ or _bootnode_ really. As every _syncing node_ are actually _bootnodes_, you can just check whether the node is a _bootnode_, and skip the `bootnode` `bool` var, yeah?

Reply:
makes sense